engine/src/main/java/org/terasology/entitySystem/entity/EntityBuilder.java

```diff
@@ -30,11 +33,59 @@
  */
 public class EntityBuilder implements MutableComponentContainer {
 
+
+    private static final Logger logger = LoggerFactory.getLogger(EntityBuilder.class);
+
     private Map<Class<? extends Component>, Component> components = Maps.newHashMap();
-    private EngineEntityManager manager;
+    private EntityPool pool;
+    private EngineEntityManager entityManager;
+
+    private boolean sendLifecycleEvents = true;
+
+    public EntityBuilder(EngineEntityManager entityManager) {
+        this.entityManager = entityManager;
+        this.pool = entityManager.getGlobalPool();
+    }
 
-    public EntityBuilder(EngineEntityManager manager) {
-        this.manager = manager;
+    public EntityBuilder(EngineEntityManager entityManager, EntityPool pool) {
+        this.entityManager = entityManager;
+        this.pool = pool;
+    }
+
+    /**
+     * Adds all of the components from a prefab to this builder
+     *
+     * @param prefabName the name of the prefab to add
+     * @return whether the prefab was successfully added
+     */
+    public boolean addPrefab(String prefabName) {
+        if (prefabName != null && !prefabName.isEmpty()) {
+            Prefab prefab = entityManager.getPrefabManager().getPrefab(prefabName);
+            if (prefab == null) {
+                logger.warn("Unable to instantiate unknown prefab: \"{}\"", prefabName);
+                return false;
+            }
+            addPrefab(prefab);
+            return true;
+        } else {
+            return false;
+        }
+    }
+    /**
+     * Adds all of the components from a prefab to this builder
+     *
+     * @param prefab the prefab to add
+     * @return whether the prefab was successfully added
+     */
+    public void addPrefab(Prefab prefab) {
```

Comment:
I think I just copied the Javadoc without properly checking it.

It currently doesn't have anywhere it can fail, though I guess passing a null prefab might cause it to return false. The addPrefab(String) one only fails because a non-null string can fail to return a prefab (though, looking at it again, it will fail and not add an EntityInfoComponent on a nulll string, so changing one of them would make them consistent).

Currently adding a null prefab is fine, and is used to create an entity when you want to add a location, but not a prefab.